src-electron/generator/matter/controller/python/templates/helper.js

```diff
@@ -43,6 +49,58 @@ function asPythonType(zclType) {
   }
 }
 
+/**
+ *
+ * Available options:
+ * - All options passed to this helper are considered as overrides for
+ * zcl types
+ * for eg: (as_underlying_python_zcl_type type clusterId boolean='bool')
+ * will return "bool" for "boolean" type
+ * @param {*} type
+ * @param {*} clusterId
+ * @param {*} options
+ * @returns The corresponding python data type for a zcl data type.
+ */
+async function as_underlying_python_zcl_type(type, clusterId, options) {
+  let hash = options.hash;
+  // Overwrite any type with the one coming from the template options
```

Comment:
This is only used for "boolean" and "BOOLEAN", right?  Why wouldn't we just handle those directly in this helper like we handle other types?

Reply:
2 things:
- It becomes a one line corresponding change in the template. So less questions asked in a PR:)
- I sometimes am not sure if this was sufficient. In case there are more use cases which I have not covered. Makes it easy to add/refer to additional exceptions.

Reply:
> It becomes a one line corresponding change in the template. So less questions asked in a PR:)

I'm not sure I follow.  The template change would be one line no matter what; we would just remove the boolean= and BOOLEAN= bits from the template and handle them here.

I think I am fine leaving the functionality, but it's weird to handle string types here, but not boolean types...

Reply:
Leaving this as is for now

Reply:
Again, we should handle booleans correctly without overrides, even if we leave the overrides in.